waspc/cli/src/Wasp/Cli/Command/CreateNewProject/AI.hs

```diff
@@ -6,32 +6,110 @@ module Wasp.Cli.Command.CreateNewProject.AI
 where
 
 import Control.Arrow ()
+import Control.Monad (unless)
 import Control.Monad.Except (MonadError (throwError), MonadIO (liftIO))
 import Data.Function ((&))
+import Data.Functor ((<&>))
+import Data.List (intercalate)
+import qualified Data.List.NonEmpty as NE
 import qualified Data.Text as T
 import qualified Data.Text.IO as T.IO
-import StrongPath (Abs, Dir, Path', fromAbsDir)
+import StrongPath (Abs, Dir, Path', basename, fromAbsDir, fromRelDir)
 import StrongPath.Operations ()
 import System.Directory (createDirectory, createDirectoryIfMissing, setCurrentDirectory)
 import System.Environment (lookupEnv)
 import System.FilePath (takeDirectory)
+import qualified System.FilePath as FP
 import System.IO (hFlush, stdout)
 import qualified Wasp.AI.CodeAgent as CA
 import qualified Wasp.AI.GenerateNewProject as GNP
-import Wasp.AI.GenerateNewProject.Common (NewProjectConfig, NewProjectDetails (..), emptyNewProjectConfig)
+import Wasp.AI.GenerateNewProject.Common
+  ( NewProjectConfig,
+    NewProjectDetails (..),
+    emptyNewProjectConfig,
+  )
+import qualified Wasp.AI.GenerateNewProject.Common as GNP.C
+import qualified Wasp.AI.GenerateNewProject.LogMsg as GNP.L
 import Wasp.AI.OpenAI (OpenAIApiKey)
+import qualified Wasp.AI.OpenAI.ChatGPT as ChatGPT
 import Wasp.Cli.Command (Command, CommandError (CommandError))
-import Wasp.Cli.Command.CreateNewProject.ProjectDescription (NewProjectAppName (..), obtainAvailableProjectDirPath, parseWaspProjectNameIntoAppName)
+import Wasp.Cli.Command.CreateNewProject.ProjectDescription
+  ( NewProjectAppName (..),
+    obtainAvailableProjectDirPath,
+    parseWaspProjectNameIntoAppName,
+  )
 import Wasp.Cli.Command.CreateNewProject.StarterTemplates (readWaspProjectSkeletonFiles)
 import Wasp.Cli.Common (WaspProjectDir)
 import qualified Wasp.Cli.Interactive as Interactive
+import qualified Wasp.Util as U
 import qualified Wasp.Util.Aeson as Utils.Aeson
+import qualified Wasp.Util.Terminal as T
 
 createNewProjectInteractiveOnDisk :: Path' Abs (Dir WaspProjectDir) -> NewProjectAppName -> Command ()
 createNewProjectInteractiveOnDisk waspProjectDir appName = do
   openAIApiKey <- getOpenAIApiKey
   appDescription <- liftIO $ Interactive.askForRequiredInput "Describe your app in a couple of sentences"
-  liftIO $ createNewProjectOnDisk openAIApiKey waspProjectDir appName appDescription emptyNewProjectConfig
+  (planningGptModel, codingGptModel) <-
+    liftIO $
+      Interactive.askToChoose'
```

Comment:
Makes sense 👍 
`
One weird thing for me is the use of `askToChoose'` vs. `askToChoose` as the exposed function. Can we change that?

Reply:
Makes sense, but both are valid choices to use -> one is for when dealing with Option, another is for more general case.

WHy not use `askToChoose` for all of them? Beacuse it would return `Option ...` and when you are using `Option`, you really want to get the value, not the whole option.

Since they are so similar, I thought `'` works fine ok, it is a specialized version of `askToChoose`.

Reply:
Okay, makes sense 👍